rdmd.d

```diff
@@ -916,3 +900,41 @@ void yap(size_t line = __LINE__, T...)(auto ref T stuff)
     debug stderr.writeln(line, ": ", stuff);
     else stderr.writeln(stuff);
 }
+
+/**
+Used to wrap filesystem operations that should also be logged with the
+yap function. These functions allow the the filenane to be given once
+so the log statement always matches the operation. Using it also guarantees
+you won't forget to include a `yap` alongside any file operation you want
+to be logged.
+*/
+static struct Yap
+{
+    static auto opDispatch(string op, T...)(T args)
+    {
+        static if (op == "exists" ||
+                   op == "timeLastModified" ||
+                   op == "isFile")
+            yap("stat ", args[0]);
+        else static if (op == "rename")
+            yap("mv ", args[0], " ", args[1]);
+        else static if (op == "remove" ||
+                        op == "rmdirRecurse" ||
+                        op == "dirEntries")
+            yap(op, " ", args[0]);
```

Comment:
This looks pretty scary and not simple at all. You can no longer understand what this does at a glance.

Reply:
I recall there was a Phobos function something like `x.isOneOf(a, b, c)` which would return the equivalent of `x==a || x==b || x==c`, but I don't remember what it was called.

Reply:
`switch` would be fine here too, I think.

Reply:
would require `static switch`

Reply:
changed to use `findAmong`

Reply:
> would require static switch

Not *really* though, right? It would just optimize out.

Reply:
The cases won't compile...so runtime switch won't work.

Reply:
i.e. `args.length` will only be 1 for some functions, but 2 for others, so you get a compile error since a runtime switch requires all cases to be compilable.

Reply:
Oh right, thanks.